### PR TITLE
Add '.gitignore' file to gp

### PR DESCRIPTION
**Pull request by Postyy**

### CODE REVIEW

File: artifactory/utils/golang/project/project.go [hunk 1]
```diff
@@ -324,11 +323,10 @@ func parseModuleName(modContent string) (string, error) {
 }
 
 // Returns a regex that match the following:
```
Comment (eyalbe4) [inline artifactory/utils/golang/project/project.go]: match --> matches

File: artifactory/utils/golang/project/project.go [hunk 2]
```diff
@@ -324,11 +323,10 @@ func parseModuleName(modContent string) (string, error) {
...
-// 1. .git folder.
-// 2. .gitignore file
-// 3. .DS_Store
+// 1. .DS_Store.
+// 2. .git.
 func getPathExclusionRegExp() (*regexp.Regexp, error) {
```
Comment (eyalbe4) [inline artifactory/utils/golang/project/project.go]: Should we also exclude files that are marked as ignored in the **.gitignore**?

Comment (Postyy) [reply artifactory/utils/golang/project/project.go]: Fixed, Ignore files according to 'gitignore' pattern

File: artifactory/utils/golang/project/project.go (hunk 1 above)
Comment (Postyy) [reply artifactory/utils/golang/project/project.go]: Fixed

**Comment by eyalbe4** [comment]
@Postyy - Id the new *new.dll* file used to check the new gitignore functionality?
If it is, then I suggest to rename the file to something like *test-go-publish-git-ignore.dll*.

File: artifactory/utils/golang/project/archive.go [hunk 3]
```diff
@@ -3,27 +3,31 @@ package project
...
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
 )
 
 // Archive project files according to the go project standard
 func archiveProject(writer io.Writer, sourcePath, module, version string, excludePathsRegExp *regexp.Regexp) error {
 	zipWriter := zip.NewWriter(writer)
 	defer zipWriter.Close()
 
+	ignore, _ := gitignore.NewFromFile(sourcePath + "/.gitignore")
```
Comment (eyalbe4) [inline artifactory/utils/golang/project/archive.go]: The error returned from this function is ignored.

File: artifactory/utils/golang/project/archive.go [hunk 4]
```diff
@@ -53,3 +57,16 @@ func getFileName(sourcePath, filePath, moduleName, version string) string {
 
 	return filepath.Join(moduleID, filename)
 }
+
+func isExcludePathsByGitignore(path string, ignore gitignore.GitIgnore) bool {
```
Comment (eyalbe4) [inline artifactory/utils/golang/project/archive.go]: I suggest renaming this function to *isPathIgnoredByGit*.

File: artifactory/utils/golang/project/archive.go (hunk 3 above)
Comment (Postyy) [reply artifactory/utils/golang/project/archive.go]: Fixed, thanks!.

File: artifactory/utils/golang/project/archive.go (hunk 4 above)
Comment (Postyy) [reply artifactory/utils/golang/project/archive.go]: Done.

**Pull request merged**